### PR TITLE
340 add code examples

### LINKED ISSUE #340: Add common program examples to code input page
As a first time player of the game

I want to be able to choose from a few basic programs

Because it enables me to check out the capabilities of the application without having to learn redcode.

NOTES:
- Maybe add Imp, Dwarf and some third program.
- Design the example program selection similar to the challenges found on https://bitshift.flamion.dev/

Dwarf:
```
add.ab  #4, 3
mov.i   2, @2
jmp     -2
dat     #0, #0
```

Mice:
```
mov.ab  #12, -1
mov.i   @-2, <5
djn     -1, -3
spl     @3, 0
add.ab  #653, 2
jmz     -5, -6
dat     #0, #833
```

Scanner:
```
add    #10, 1
jmz.f  -1, 5
mov.i  2, >-1
jmp    -1
```

Acceptance Criteria:
- [ ] 3 cards exist next to the code editor
- [ ] The cards have the title of the program they represent and a short description of what the program does.
- [ ] Each card has a button at the bottom, that opens a toast, asking the user if they want to select this program, overriding their current program.
- [ ] Pressing Ok in the toast, replaces the current program with the selected program.
- [ ] The code is fully tested or a good explanation given why it should not or cannot be tested

### PR DESCRIPTION
**Pull request by Straczi**
closes #340

### CODE REVIEW

**Comment by Straczi** [comment]
currenty branched from game visu branch, once that one is merged, this PR will be opened :+1:

**Review by 0815Sailsman — CHANGES_REQUESTED** [review]
The page is now scrollable. Would prefer it not exceed the window height

Also, as seen in the following image, the code editor itself extends too far to the bottom. IMO either dont force the example cards and code editor to have the same height, or just extend the actual code editor aswell.
image

**Comment by Straczi** [comment]
> The page is now scrollable. Would prefer it not exceed the window height
> 
> Also, as seen in the following image, the code editor itself extends too far to the bottom. IMO either dont force the example cards and code editor to have the same height, or just extend the actual code editor aswell. image

hm, i styled this on a 1080p monitor and it looked fine and it still looks fine on my laptop :thinking: , what res do you currently use ?
Screenshot from 2024-12-16 16-43-59

**Comment by 0815Sailsman** [comment]
> > The page is now scrollable. Would prefer it not exceed the window height
> > Also, as seen in the following image, the code editor itself extends too far to the bottom. IMO either dont force the example cards and code editor to have the same height, or just extend the actual code editor aswell. image
> 
> hm, i styled this on a 1080p monitor and it looked fine and it still looks fine on my laptop 🤔 , what res do you currently use ? Screenshot from 2024-12-16 16-43-59

1920x1080, browser used is firefox

File: frontend/app/src/lib/CodeExamples.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { CodeExample } from "@/domain/CodeExample";
+
+export const codeExamples: CodeExample[] = [
+	{
+		title: "Dwarf",
+		shortDescription:
+			'The dwarf "bombs" the core at regular intervals with DATs while making sure to not hit itself',
+		code: "add.ab  #4, 3\nmov.i   2, @2\njmp     -2\ndat     #0, #0",
```
Comment (flamion) [inline frontend/app/src/lib/CodeExamples.ts:8]: Minor nitpick, but can you add a newline at the end of the code examples? Looks a bit nicer in the editor then and you can immediately start editing on a fresh line

Comment (Straczi) [reply frontend/app/src/lib/CodeExamples.ts:8]: :nerd_face:

Comment (Straczi) [reply frontend/app/src/lib/CodeExamples.ts:8]: will be done :)

Comment (Straczi) [reply frontend/app/src/lib/CodeExamples.ts]: done :+1:

**Review by 0815Sailsman — APPROVED** [review]
seems like the layouting issue is on my side, so feel free to merge

**Pull request merged**